examples/nextjs-example/pages/index.tsx

```diff
@@ -33,9 +33,26 @@ const TenantLabels = {
 
 export default function Home() {
   const { userId, isLoading, userToken } = useIdentify();
+  const [currentUserToken, setCurrentUserToken] = useState(null);
   const [tenant, setTenant] = useState(Tenants.TeamA);
 
-  if (isLoading || !userToken) {
+  useEffect(() => {
+    if (userToken) {
+      setCurrentUserToken(userToken);
+    }
+  }, [userToken]);
+
+  const tokenRefreshHandler = useCallback(async () => {
+    // Refresh the user token 1s before it expires
+    const res = await fetch(`/api/auth?id=${userId}`);
+    const json = await res.json();
+
+    setCurrentUserToken(json.userToken);
+
+    return;
+  }, [userId, setCurrentUserToken]);
```

Comment:
Could you explain a little more why we need `currentUserToken`? Would it make more sense for the example to have `setUserToken` come from `useIdentify` instead? 

I also might just be missing something here? If that's the case it might make sense to write a comment explaining what's going on to the engineer viewing this example.

Reply:
Oh yeah, that's a great point. Let me clean this up actually, I was just trying out a different way to manage setting this userToken so on refresh the value was consistent .

Reply:
Nice!